dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -613,53 +421,70 @@ export default defineComponent({
       try {
         this.postEuTaxonomyDataForNonFinancialsProcessed = false;
         this.messageCount++;
-        const formInputsModelToSend = modifyObjectKeys(
-          JSON.parse(JSON.stringify(this.formInputsModel)) as objectType,
-          "send"
+
+        checkIfAllUploadedReportsAreReferencedInDataModel(
+          this.formInputsModel.data as ObjectType,
+          this.namesOfAllCompanyReportsForTheDataset
         );
+        await (this.$refs.UploadReports.uploadFiles as () => Promise<void>)();
+
+        await this.$nextTick();
+        const formInputsModelToSend = modifyObjectKeys(this.formInputsModel as ObjectType, "send");
         const euTaxonomyDataForNonFinancialsControllerApi = await new ApiClientProvider(
           assertDefined(this.getKeycloakPromise)()
         ).getEuTaxonomyDataForNonFinancialsControllerApi();
         this.postEuTaxonomyDataForNonFinancialsResponse =
           await euTaxonomyDataForNonFinancialsControllerApi.postCompanyAssociatedEuTaxonomyDataForNonFinancials(
-            formInputsModelToSend
+            formInputsModelToSend as CompanyAssociatedDataEuTaxonomyDataForNonFinancials
           );
         this.$emit("datasetCreated");
-        this.$formkit.reset(this.formId);
       } catch (error) {
-        this.postEuTaxonomyDataForNonFinancialsResponse = null;
+        this.messageCount++;
```

Comment:
Split this out into a separate function. Same for its sibling in the financials dataset creation.

Reply:
The task was done by @pioSko2, right?

Reply:
Yes. Done.